File: daily/1030.距离顺序排列矩阵单元格.js
/*
 * @lc app=leetcode.cn id=1030 lang=javascript
 *
 * [1030] 距离顺序排列矩阵单元格
 */

// @lc code=start
/**
 * @param {number} R
 * @param {number} C
 * @param {number} r0
 * @param {number} c0
 * @return {number[][]}
 */
var allCellsDistOrder = function(R, C, r0, c0) {
  const result = []
  const calc = (a) => Math.abs(a[0] - r0) + Math.abs(a[1] - c0)
  for(let r = 0; r < R; ++r)
    for(let c = 0; c < C; ++c)
      result.push([r, c])
  return result.sort((a, b) => calc(a) - calc(b))
};
// @lc code=end

File: daily/1030.距离顺序排列矩阵单元格.rs
/*
 * @lc app=leetcode.cn id=1030 lang=rust
 *
 * [1030] 距离顺序排列矩阵单元格
 */

// @lc code=start
impl Solution {
    pub fn all_cells_dist_order(r: i32, c: i32, r0: i32, c0: i32) -> Vec<Vec<i32>> {
        let mut result = Vec::new();
        (0..r).for_each(|i| 
            (0..c).for_each(|j| 
                result.push(((i - r0).abs() + (j - c0).abs(), vec![i, j]))));
        result.sort_unstable_by_key(|(v, t)| *v);
        result.into_iter().map(|v| v.1).collect()
    }
}
// @lc code=end
